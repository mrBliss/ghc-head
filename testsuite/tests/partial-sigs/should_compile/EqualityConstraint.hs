{-# LANGUAGE TypeFamilies #-}
module EqualityConstraint where

foo :: a ~ Bool => (a, _)
foo = (True, False)

test :: (Bool, Bool)
test = foo
