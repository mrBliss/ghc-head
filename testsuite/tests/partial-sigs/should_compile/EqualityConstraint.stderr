TYPE SIGNATURES
  foo :: forall a. a ~ Bool => (a, Bool)
TYPE CONSTRUCTORS
COERCION AXIOMS
Dependent modules: []
Dependent packages: [base, ghc-prim, integer-gmp]
